src/node_perf.cc

```diff
@@ -489,29 +489,53 @@ ELDHistogram::ELDHistogram(
                           Histogram(1, 3.6e12),
                           resolution_(resolution) {
   MakeWeak();
-  uv_timer_init(env->event_loop(), &timer_);
+
+  // Since we pass `timer_` to `HandleWrap` constructor - we have to
+  // initialize it here. It is equally important to have it initialized for
+  // correct operation of `Close()` below.
```

Comment:
If you turn the above lines into

```cpp
ELDHistogram::ELDHistogram(
    Environment* env,
    Local<Object> wrap,
    int32_t resolution)
  : HandleWrap(env,
               wrap,
               resolution > 0 ?
                   reinterpret_cast<uv_handle_t*>(&timer_) :
                   reinterpret_cast<uv_handle_t*>(&prepare_),
               AsyncWrap::PROVIDER_ELDHISTOGRAM),
    Histogram(1, 3.6e12),
    resolution_(resolution) {
  ...
```

Then you only need to initialize one handle, and don’t need the custom `::Close()` override … and, going one step further, you could even save some memory and turn the `timer_`/`prepare_` fields into e.g.

```cpp
union {
  uv_handle_t handle;
  uv_timer_t timer;
  uv_prepare_t prepare;
} handle_;